lib/compile/util.js

```diff
@@ -214,20 +214,33 @@ function getPath(currentPath, prop, jsonPointers) {
 }
 
 
+var JSON_POINTER = /^#|\/(?:[^~]|~0|~1)*?$/;
```

Comment:
Oh jeeze, I had never actually noticed that $refs are also JSON pointers. Now that you've pointed that out, I have noticed something. A $data reference formatted like a URI (`$data: #/foo`) will be considered valid in the schema, but the data won't validate. Formatting $ref as a non-URI pointer has a similar effct: `$ref: /definitions/foo` won't throw an error in the schema, but the data won't validate. 

According to the JSON-schema spec, a $ref has to be formatted as a URI, and the $data proposal doesn't mention URI support at all (while I can think of some uses for it, it sounds like a potential security nightmare).